instructure/canvas-ios
# Fix issue of user not being able to submit assignment when logged in root account

refs: [MBL-18064](https://instructure.atlassian.net/browse/MBL-18064)
affects: Student,Teacher
release note: Fixes issue of user not being able to submit assignment when logged in root account

## What's changed

The main point of change of this PR resides in files:
[`AppEnvironmentOverride`](https://github.com/instructure/canvas-ios/pull/2989/files#diff-2151f2282f4d1aadb4e9ae981de9758a1f7355399f3f146e18e87e53a1b8d683)
[`CourseTabUrlInteractor`](https://github.com/instructure/canvas-ios/blob/9c831a1c7135a0724efaeb85b48161ad9d776906/Core/Core/Router/CourseTabUrlInteractor.swift#L117)
[`Router`](https://github.com/instructure/canvas-ios/blob/9c831a1c7135a0724efaeb85b48161ad9d776906/Core/Core/Router/Router.swift#L348)

This is to provide a clean way to pass a modified `AppEnvironment` with the correct baseURL that fits the current tab user is sitting at, which in this case is `Course > Assignments`

The rest are a consequence of those changes, which is basically to make sure views and classes are respecting what ever `AppEnvironment` instance is passed to them.

## Test Plan

See [ticket](https://instructure.atlassian.net/browse/MBL-18064) description.

Also, made testing for ordinary accounts for following use cases:
1. View Assignment details
2. View Submission details
3. New text and file assignment submission.
4. New text and file submission comment.
5. View submission comments, especially those with comments.

Also, testing for the following scenarios:
1. Tapping on external link.
2. Tapping on an internal link for an assignment, making sure submission work as well for both of cases.

## Screen Record

https://github.com/user-attachments/assets/4b0c8fe8-f8de-48a3-9789-e6eafdc8a047

## Checklist



[MBL-18064]: https://instructure.atlassian.net/browse/MBL-18064?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
